src/libraries/Common/src/Interop/Windows/NtDll/Interop.NtCreateFile.cs

```diff
@@ -67,6 +76,122 @@ internal static partial class NtDll
             }
         }
 
+        internal static unsafe (uint status, IntPtr handle) CreateFile(ReadOnlySpan<char> path, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize)
+        {
+            // For mitigating local elevation of privilege attack through named pipes
+            // make sure we always call NtCreateFile with SECURITY_ANONYMOUS so that the
+            // named pipe server can't impersonate a high privileged client security context
+            SECURITY_QUALITY_OF_SERVICE securityQualityOfService = new SECURITY_QUALITY_OF_SERVICE(
```

Comment:
@carlossanlop @Jozkee this is to mimic the following logic:

https://github.com/dotnet/runtime/blob/88211b9a1fe17819e9cabf78bfd86f6d62228034/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs#L83-L87